weld/partial_types.rs

```diff
@@ -10,6 +10,7 @@ use super::error::*;
 pub enum PartialType {
     Unknown,
     Scalar(ScalarKind),
+    Vectorized(ScalarKind),
     Vector(Box<PartialType>),
```

Comment:
As a nit, I think this Vectorized vs Vector is kind of confusing. Not sure how to change it except to change Vector to List or Array. Maybe we can make that change later so we don't delay this PR, but think about alternative names here.